### ISSUE TITLE
arrange() not working before compute()

### ISSUE THREAD

**Issue by bobbymc0** [sparklyr/sparklyr#3151]
### Issue

I get the following error with dplyr::arrange() function in sparklyr 1.7.0 and 1.7.1.
It works with sparklyr 1.6.3.

`
Error in strsplit(y, "\\.")[[1]] : subscript out of bounds
`

### Reproducible example

```
# libraries + spark_connect()
library(dplyr); library(sparklyr)
sc = spark_connect(master = "local")

# simple spark dataframe
sdf = data.frame("var1" = c("a", "a", "a", "b", "b"),
                 "value" = 5:1) %>%
  sdf_copy_to(sc = sc, name = "sdf", overwrite = TRUE)

# arrange() not working
sdf %>%
  arrange(value) %>%
  compute()

```

### sessionInfo()
```
sessionInfo()
R version 4.0.2 (2020-06-22)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: OS X  11.4

Matrix products: default
LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] sparklyr_1.7.1 dplyr_1.0.7   

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.6         pillar_1.6.1       compiler_4.0.2     later_1.2.0        dbplyr_2.1.1       r2d3_0.2.5         base64enc_0.1-3   
 [8] shinyjs_2.0.0      tools_4.0.2        digest_0.6.27      jsonlite_1.7.2     lifecycle_1.0.0    tibble_3.1.2       pkgconfig_2.0.3   
[15] rlang_0.4.11       rstudioapi_0.13    shiny_1.6.0        DBI_1.1.1          parallel_4.0.2     yaml_2.2.1         fastmap_1.1.0     
[22] withr_2.4.2        httr_1.4.2         askpass_1.1        rappdirs_0.3.3     shinyEffects_0.2.0 generics_0.1.0     vctrs_0.3.8       
[29] htmlwidgets_1.5.3  rprojroot_2.0.2    tidyselect_1.1.0   glue_1.4.2         forge_0.2.0        data.table_1.14.0  R6_2.5.0          
[36] fansi_0.5.0        tidyr_1.1.3        purrr_0.3.4        magrittr_2.0.1     promises_1.2.0.1   ellipsis_0.3.2     htmltools_0.5.1.1 
[43] assertthat_0.2.1   mime_0.10          xtable_1.8-4       httpuv_1.6.1       config_0.3.1       utf8_1.2.1         openssl_1.4.4     
[50] crayon_1.4.1      

```

**Comment by yitao-li** [comment]
@bobbymc0 This was addressed recently. I believe it was due to some change in `dbplyr` rather than changes in `sparklyr` itself. 

You can run `remotes::install_github("yitao-li/sparklyr", ref = "bugfix/sdf-remote-name-for-op-order")` and restart your R session to install a version of `sparklyr` that is free from this issue.

**Comment by nviets** [comment]
This error appears to have returned as of 1.7.0 and 1.7.1.

```
> library(dplyr); library(sparklyr)
> sc = spark_connect(master = "local")
>
> # simple spark dataframe
> sdf = data.frame("var1" = c("a", "a", "a", "b", "b"),
+                  "value" = 5:1) %>%
+   sdf_copy_to(sc = sc, name = "sdf", overwrite = TRUE)
>
> # arrange() not working
> sdf %>%
+   arrange(value) %>%
+   compute()
Error in strsplit(y, "\\.")[[1]] : subscript out of bounds
```

**Comment by yitao-li** [comment]
@nviets It was definitely fixed in `sparklyr` 1.7.2 (https://cloud.r-project.org/src/contrib/sparklyr_1.7.2.tar.gz). Try `install.packages("sparklyr")` to install the latest version of `sparklyr`.
